src/lib/snack-bar/snack-bar.spec.ts

```diff
@@ -259,6 +260,55 @@ describe('MdSnackBar', () => {
     // Let remaining animations run.
     tick(500);
   }));
+
+  it('should dismiss the snackbar when the action is called, notifying of both action and dismiss',
+     fakeAsync(() => {
+       let dismissObservableCompleted = false;
+       let actionObservableCompleted = false;
+       let snackBarRef = snackBar.open('Some content', 'dismiss');
+       viewContainerFixture.detectChanges();
+
+       snackBarRef.afterDismissed().subscribe(null, null, () => {
+         dismissObservableCompleted = true;
+       });
+       snackBarRef.onAction().subscribe(null, null, () => {
+         actionObservableCompleted = true;
+      });
+
+      let actionButton =
+        overlayContainerElement.querySelector('.md-simple-snackbar-action') as HTMLButtonElement;
+      actionButton.click();
+      viewContainerFixture.detectChanges();
+      flushMicrotasks();
+
+      viewContainerFixture.whenStable().then(() => {
+        expect(dismissObservableCompleted).toBeTruthy('Expected the snack bar to be dismissed');
+        expect(actionObservableCompleted).toBeTruthy('Expected the snack bar to notify of action');
+      });
+
+      tick(500);
+    }));
+
+    it('should dismiss automatically after a specified timeout', fakeAsync(() => {
+      let dismissObservableCompleted = false;
+      let config = new MdSnackBarConfig();
+      config.autoHide = 250;
+      let snackBarRef = snackBar.open('content', 'test', config);
+      snackBarRef.afterDismissed().subscribe(null, null, () => {
+        dismissObservableCompleted = true;
+      });
+
+      viewContainerFixture.detectChanges();
+      viewContainerFixture.whenStable().then(() => {
+        expect(dismissObservableCompleted).toBeFalsy('Expected the snack bar not to be dismissed');
+      });
+
+      tick(1000);
+      viewContainerFixture.detectChanges();
+      viewContainerFixture.whenStable().then(() => {
```

Comment:
You shouldn't need to use `whenStable` with `fakeAsync` because the `whenStable` will just be called synchronously. 

(same for other uses)